### PR TITLE
web: add app mount shared span

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

This PR adds another shared span — `AppMount`- representing the React application mount event. It will be used to calculate time from React app mount until the target React component mount. E.g., how much time on average do we wait for the `Blob` component to mount after `SourcegraphWeb` did mount?

Partially addresses https://github.com/sourcegraph/sourcegraph/issues/44008.

## Test plan

1. `ENABLE_OPEN_TELEMETRY=true sg start web-standalone`
2. Open the browser console and ensure that debug-level output is enabled.
3. Reload any web application page.
4. See the app mount span attached to the window-load span.

Screenshot 2022-11-07 at 16 42 30

## App preview:

- [Web](https://sg-web-vb-add-app-mount-span.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.05% (+1.43 kb) | 0.01% (+1.44 kb) | 0.00% (+0.01 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vb/add-app-mount-span/compare-report.html;tab=live_object) for a full comparison between the commits 7c41ac8 and 5773b39 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/components/ComponentsComposer.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *   <ComponentsComposer contexts={[
+ *      <WildcardThemeContext.Provider key="wildcard" value={WILDCARD_THEME} />,
+ *      <ErrorBoundary key="error-boundary" location={null} />,
+ *   ]}>
+ *       <main>My app</main>
+ *       <footer />
+ *   </ComponentsComposer>
+ * ```
+ *
+ */
```
Comment (philipp-spiess) [inline client/web/src/components/ComponentsComposer.tsx:22]: Amazing cleanup!

File: client/web/src/SourcegraphWebApp.tsx [hunk 2]
```diff
@@ -356,97 +358,77 @@ export class SourcegraphWebApp extends React.Component<
...
-                </WildcardThemeContext.Provider>
-            </ApolloProvider>
+            <ComponentsComposer
+                components={[
+                    // `ComponentsComposer` provides children via `React.cloneElement`.
+                    /* eslint-disable react/no-children-prop, react/jsx-key */
+                    <ApolloProvider client={graphqlClient} children={undefined} />,
+                    <WildcardThemeContext.Provider value={WILDCARD_THEME} />,
+                    <ErrorBoundary location={null} />,
+                    <TraceSpanProvider name={SharedSpanName.AppMount} />,
```
Comment (philipp-spiess) [inline client/web/src/SourcegraphWebApp.tsx:368]: 👀

**Review by philipp-spiess — APPROVED** [review]

File: client/web/src/SourcegraphWebApp.tsx [hunk 3]
```diff
@@ -356,97 +358,77 @@ export class SourcegraphWebApp extends React.Component<
...
+                    <FeatureFlagsProvider />,
+                    <ShortcutProvider />,
+                    <TemporarySettingsProvider temporarySettingsStorage={temporarySettingsStorage} />,
+                    <SearchResultsCacheProvider />,
+                    <SearchQueryStateStoreProvider useSearchQueryState={useNavbarQueryState} />,
+                    /* eslint-enable react/no-children-prop, react/jsx-key */
+                ]}
+            >
+                <Router history={history} key={0}>
```
Comment (vovakulikov) [inline client/web/src/SourcegraphWebApp.tsx:377]: Can the router be placed inside of the ComponentsComposer components list?

**Review by vovakulikov — APPROVED** [review]

File: client/web/src/SourcegraphWebApp.tsx (hunk 3 above)
Comment (valerybugakov) [reply client/web/src/SourcegraphWebApp.tsx:377]: We have multiple children passed to the `ComponentsComposer` component currently. Adding `Router` to the `ComponentsComposer` components list requires changing that.

Screenshot 2022-11-08 at 10 02 25

**Pull request merged**